src/interactivity/base/HostSignalInputThread.cpp

```diff
@@ -0,0 +1,255 @@
+//      (in and screen buffers) haven't yet been initialized.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void HostSignalInputThread::ConnectConsole() noexcept
+{
+    _consoleConnected = true;
+}
+
+// Method Description:
+// - The ThreadProc for the Host Signal Input Thread.
+// Return Value:
+// - S_OK if the thread runs to completion.
+// - Otherwise it may cause an application termination another route and never return.
+[[nodiscard]] HRESULT HostSignalInputThread::_InputThread()
+{
+    unsigned short signalId;
+    while (_GetData(&signalId, sizeof(signalId)))
+    {
+        switch (signalId)
+        {
+        case HOST_SIGNAL_NOTIFY_APP:
+        {
+            HOST_SIGNAL_NOTIFY_APP_DATA msg = { 0 };
+            if (_GetData(&msg, sizeof(msg)))
+            {
+                if (msg.cbSize >= sizeof(msg))
+                {
+                    _AdvanceReader(msg.cbSize - sizeof(msg));
+                }
+                else
+                {
+                    THROW_HR(E_ILLEGAL_METHOD_CALL);
+                }
+
+                LOG_IF_NTSTATUS_FAILED(ServiceLocator::LocateConsoleControl()->NotifyConsoleApplication(msg.dwProcessId));
+            }
+            else
+            {
+                return E_ABORT;
+            }
+
+            break;
+        }
+        case HOST_SIGNAL_SET_FOREGROUND:
+        {
+            HOST_SIGNAL_SET_FOREGROUND_DATA msg = { 0 };
+            if (_GetData(&msg, sizeof(msg)))
+            {
+                if (msg.cbSize >= sizeof(msg))
+                {
+                    _AdvanceReader(msg.cbSize - sizeof(msg));
+                }
+                else
+                {
+                    THROW_HR(E_ILLEGAL_METHOD_CALL);
+                }
+
+                LOG_IF_NTSTATUS_FAILED(ServiceLocator::LocateConsoleControl()->SetForeground(ULongToHandle(msg.dwProcessId), msg.fForeground));
+            }
+            else
+            {
+                return E_ABORT;
+            }
+            break;
+        }
+        case HOST_SIGNAL_END_TASK:
+        {
+            HOST_SIGNAL_END_TASK_DATA msg = { 0 };
+            if (_GetData(&msg, sizeof(msg)))
+            {
+                if (msg.cbSize >= sizeof(msg))
+                {
+                    _AdvanceReader(msg.cbSize - sizeof(msg));
+                }
+                else
+                {
+                    THROW_HR(E_ILLEGAL_METHOD_CALL);
+                }
+
+                LOG_IF_NTSTATUS_FAILED(ServiceLocator::LocateConsoleControl()->EndTask(ULongToHandle(msg.dwProcessId), msg.dwEventType, msg.ulCtrlFlags));
+            }
+            else
+            {
+                return E_ABORT;
+            }
+            break;
+        }
+        default:
+        {
+            THROW_HR(E_UNEXPECTED);
+            break;
+        }
+        }
+    }
+    return S_OK;
+}
+
+// Method Description:
+// - Skips the file stream forward by the specified number of bytes.zs

```

Comment:
a bunch of `.zs` appeared. interesting.

```suggestion
// - Skips the file stream forward by the specified number of bytes.
```

Reply:
That's a keyboard thing. I'm trying to hit Ctrl+S to save but I'm not waiting long enough for the Ctrl to roll over out of the Z key. Oops.